cellprofiler/utilities/hdf5_dict.py

```diff
@@ -1024,6 +1024,12 @@ def __init__(self, hdf5_file, lock=None, filelist_name=DEFAULT_GROUP):
         g = self.hdf5_file.require_group(FILE_LIST_GROUP)
         if filelist_name in g:
             g = g[filelist_name]
+            # In Python 2, some hdf5 vals are stored as bytes. Convert to Python 3 unicode string
+            if g.attrs.get(A_CLASS, None):
+                try:
+                    g.attrs[A_CLASS] = g.attrs[A_CLASS].decode()
+                except AttributeError:
```

Comment:
What is this case handling?

Reply:
when loading CP 3.1.8 cpproj files, `g.attrs[A_CLASS]` will be a `bytes` object, and because it's being compared to a `str`, it's causing the `assert` below to fail.

Reply:
I ended up scrapping this change and added an equality checker at `cellprofiler.utilities.legacy.equals()` to replace the `==` when comparing between bytes and strings. Thoughts?